tgui/packages/tgui/interfaces/Radio.js

```diff
@@ -0,0 +1,127 @@
+import { map } from 'common/collections';
+import { toFixed } from 'common/math';
+import { useBackend } from '../backend';
+import { Box, Button, Fragment, LabeledList, NumberInput, Section } from '../components';
+import { RADIO_CHANNELS } from '../constants';
+import { Window } from '../layouts';
+
+export const Radio = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    freqlock,
+    frequency,
+    minFrequency,
+    maxFrequency,
+    canReset,
+    listening,
+    broadcasting,
+    loudspeaker,
+    has_loudspeaker,
+    subspace,
+    subspaceSwitchable,
+  } = data;
+  const tunedChannel = RADIO_CHANNELS
+    .find(channel => channel.freq === frequency);
+  const channels = map((value, key) => ({
+    name: key,
+    status: !!value,
+  }))(data.channels);
+  return (
+    <Window>
+      <Window.Content>
+        <Section>
+          <LabeledList>
+            <LabeledList.Item label="Frequency">
+              {freqlock && (
+                <Box inline color="light-gray">
+                  {toFixed(frequency / 10, 1) + ' kHz'}
+                </Box>
+              ) || (
+                <Fragment>
+                  <NumberInput
+                    animate
+                    unit="kHz"
+                    step={0.2}
+                    stepPixelSize={10}
+                    minValue={minFrequency / 10}
+                    maxValue={maxFrequency / 10}
+                    value={frequency / 10}
+                    format={value => toFixed(value, 1)}
+                    onDrag={(e, value) => act('frequency', {
```

Comment:
Use `onChange` here so it only sends the topic to DD when the user lets go, instead of on each increment.